s3_storage/models/s3storage.py

```diff
@@ -1,113 +1,110 @@
+"""ir.attachment extension"""
 # coding: utf-8
 ##############################################################################
-#    This file is part of the fast_fetchmail module.
+#    This file is part of the s3_storage module.
 #
-#    fast_fetchmail is free software: you can redistribute it and/or modify
+#    s3_storage is free software: you can redistribute it and/or modify
 #    it under the terms of the GNU Lesser General Public License as published
 #    by the Free Software Foundation, either version 3 of the License, or
 #    (at your option) any later version.
 #
-#    fast_fetchmail is distributed in the hope that it will be useful,
+#    s3_storage is distributed in the hope that it will be useful,
 #    but WITHOUT ANY WARRANTY; without even the implied warranty of
 #    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 #    GNU Lesser General Public License for more details.
 #
 #    You should have received a copy of the GNU Lesser General Public License
-#    along with fast_fetchmail. If not, see <http://www.gnu.org/licenses/>.
+#    along with s3_storage. If not, see <http://www.gnu.org/licenses/>.
 ##############################################################################
 from cStringIO import StringIO
 import logging
 
 from openerp import api, models, fields
 from openerp.tools import human_size
 
-_logger = logging.getLogger(__name__)
+LOGGER = logging.getLogger(__name__)
 try:
     from minio import Minio
     from minio.error import ResponseError
 except ImportError:
-    _logger.error('minio is required')
-
-
-class S3AttachmentsToDelete(models.Model):
-    _name = 's3.garbage'
-
-    fname = fields.Text()
+    LOGGER.error('minio package is required to store attachments on S3')
 
 
 class S3Attachment(models.Model):
+    """ir.attachment extension"""
     _inherit = 'ir.attachment'
 
+    @api.model
+    def _storage(self):
+        """
+        Get the storage type.
+        In case of missing S3 configurations, the storage defaults to file
+        """
+        res = super(S3Attachment, self)._storage()
+        if res == 's3':
+            params = ['s3_attachment_endpoint', 's3_attachment_access_key',
+                      's3_attachment_secret_key', 's3_attachment_bucket']
+            config = self.env['ir.config_parameter'].sudo()
+            for param in params:
+                if not config.get_param(param, False):
+                    res = 'file'
+                    LOGGER.error('Missing S3 configuration: %s', param)
+                    break
+        return res
+
     @api.model
     def _get_storage_client(self):
         config = self.env['ir.config_parameter'].sudo()
+        secure = config.get_param('s3_attachment_secure', True)
```

Comment:
Since you're adding a new key, it'd be nice to update the README.

Reply:
Sorry I forgot. added on 7374dab